### PR TITLE
Training Reactions Added and New Groups Defined in R_Addition_Multiplebond

### PR DESCRIPTION
**Pull request by laitcl**
Four Commits to explain:

Commit 1:
The R_Addition_MultipleBond  family does not contain any groups that utilize the benzene bond type. I copied a several of these groups from Zach's Branch, named Aromatics_updates_3

Commit 2:
In Zach's Aromatics_updates_3 branch, groups specific to benzene bonds use single-double bond notation. I've changed these groups to use Cb bonds.

Commit 3:
Five training reactions were added to the R_Addition_MultpleBond Family. These are:
Hexylbenzene + H --> CCCCCCC1C=C[CH]C=C1
Toluene + H --> CC1[CH]C=CCC=1
Toluene + H --> CC1=C[CH]C=CC1
Toluene + H --> CC1C=CC[CH]C=1
Toluene + H --> CC1C=C[CH]C=C1

Commit 4:
I apologize for the sloppy commit header. I kind of forgot what family I was working with.

Commit message says a lot, but a resummary here:

1. All bonds in the training reaction dictionary now utilizes benzene bond notation instead of single-double bonds, so that they would fall under the correct benzene bond nodes.
2. Fixed two species in the training reaction dictionary where adjacency lists weren't labelled with *1 and *2. The species are CH2CH2NH2 and CH2NH2.
3. Added a few groups and redefined some to ensure that the training reactions I added will fall in the correct place, and other training reactions that use different compounds will not share the same group as what I had intended. 

There are some known issues outlined in the 4th commit that I am hoping will fix itself when the duplicate reactions problem is solved by Max.

### CODE REVIEW

**Comment by mliu49** [comment]
@laitcl, I fixed some of the groups, and addressed my other comment. I can help you squash some of these commits later.

**Comment by zjburas** [comment]
@laitcl did you take all of the R_Add_Multi groups from #207 ?

**Comment by laitcl** [comment]
Squashed Commits.

**Comment by mliu49** [comment]
I'm not sure if Lawrence has already answered your question, but it seems that he only took the Cb_Cb branch of the tree. Groups not in that part of the tree are not in this pull request.

**Comment by zjburas** [comment]
Yes, I discussed with @laitcl and we agreed that I could piggyback on this PR to include all of the other R_Add_Multi groups from #207 . Is that ok with you @mliu49 ? I can do it over the next couple of days.

**Comment by mliu49** [comment]
Sure, of course!

**Comment by zjburas** [comment]
Alright, all of the R_Add_MultiBond groups from #207 are now in this PR. I will go ahead and add the relevant training reactions next.

File: input/kinetics/families/R_Addition_MultipleBond/groups.py [hunk 1]
```diff
@@ -17192,6 +17192,69 @@
     kinetics = None,
 )
 
+entry(
+    index = 1071,
+    label = "Cb-H_Cb-H_o_ketene",
```
Comment (mliu49) [inline input/kinetics/families/R_Addition_MultipleBond/groups.py]: I think this group and `Cb-benzofuranNde_Cb-benzofuranDe` below were already added.

Comment (mliu49) [reply input/kinetics/families/R_Addition_MultipleBond/groups.py]: An easy fix would just be adding a new commit removing this two groups, which can be squashed into this commit later.

Comment (zjburas) [reply input/kinetics/families/R_Addition_MultipleBond/groups.py]: Done

**Comment by zjburas** [comment]
I'm trying to add the training reactions now, but it appears that the convertKineticsLibraryToTrainingReactions.ipynb does not react benzene bonds. Can we discuss sometime @mliu49 ?

Also, is it ok if I rebase this  branch on #222, which might require  a force push?

**Comment by zjburas** [comment]
I should revise the statement above: the script does not react benzene bonds in reverse. In other words, if the kinetic library includes a beta-scission reaction that forms an aromatic ring, then no match will be found by the script.

**Comment by zjburas** [comment]
I've rebased on #222 after discussing with @mliu49 . All that's left for this PR are the training reactions, which shouldn't take long to add once convertKineticsLibraryToTrainingReactions.ipynb is working as intended.

File: input/kinetics/families/R_Addition_MultipleBond/groups.py [hunk 2]
```diff
@@ -16606,9 +16610,323 @@
...
+4    H  u0 {2,S}
+5    Cb u0 {1,B}
+""",
+    kinetics = None,
+)
+
+entry(
+    index = 1074,
+    label = "Cb-H_Cb-C-Ortho",
+    group =
```
Comment (zjburas) [inline input/kinetics/families/R_Addition_MultipleBond/groups.py]: @laitcl , I think this group would be more appropriately named as Cb-H-Ortho_Cb-C.

Similarly, Cb-H_Cb-H-Meta should be Cb-H-Meta_Cb-H and Cb-H_Cb-H-Para should be Cb-H-Para_Cb-H, unless I'm misunderstanding  something.

**Comment by zjburas** [comment]
87 R_Addition_MultipleBond training reactions have been added from the aromatic libraries in #222 (as well as a couple that had still not been input from the five existing CPD libraries). I've checked that they all match a reasonable reaction template. 

There was only one training reaction that I couldn't add in the First_to_Second_Aromatic_Ring/2017_Mebel_C6H5C2H2_C2H2_highP kinetic library. @mliu49 , can you take a look? When you run the library through the convertKineticslibrarytotrainingreaction.ipynb you will see which reaction I'm talking about (H addition to naphthalene). 

Also, @laitcl , can you address the comment above regarding the naming of some of your new kinetic groups?

Once these last couple of issues have been resolved, I think this PR is ready to go.

**Comment by laitcl** [comment]
@zjburas, your naming scheme does make more sense. Please make your suggested change. Thanks

**Comment by zjburas** [comment]
Alright, other than the one problematic reaction in First_to_Second_Aromatic_Ring/2017_Mebel_C6H5C2H2_C2H2_highP I think this PR is ready for review/merging.

**Comment by mliu49** [comment]
I added the problematic reaction. The issue was with different resonance structures when trying to filter by products. I've added a fix to ReactionMechanismGenerator/RMG-Py#1219.

On a related note, there seems to be many duplicates in the training reactions. In the case of the problematic reaction, there are actually 3 entries for the same reaction: forwards and backwards rates for on radical position, then the forward rate for a different radical position. There likely isn't any harm in having duplicate entries, but I don't think they necessarily add anything either since we only use the rate from the first entry for a particular template. Does anyone have thoughts on this?

**Comment by mliu49** [comment]
I've rebased this PR, squashed a few of the commits together, and added two new commits. I plan on merging this once the issues with Travis are sorted out, unless anyone has further comments.

**Pull request merged**